homeassistant/components/essent/sensor.py

```diff
@@ -0,0 +1,101 @@
+"""Support for Essent API."""
+
+import xml.etree.ElementTree as ET
+
+from pyessent import PyEssent
+
+from homeassistant.const import ENERGY_KILO_WATT_HOUR, STATE_UNKNOWN
+from homeassistant.helpers.entity import Entity
+
+DOMAIN = 'essent'
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Essent platform."""
+    username = config['username']
+    password = config['password']
+    for meter in EssentBase(username, password).retrieve_meters():
+        for tariff in ['L', 'N', 'H']:
+            add_devices([EssentMeter(username, password, meter, tariff)])
+
+
+class EssentBase():
+    """Essent Base."""
+
+    def __init__(self, username, password):
+        """Initialize the Essent API."""
+        self._essent = PyEssent(username, password)
+
+    def get_session(self):
+        """Return the active session."""
+        return self._essent
+
+    def retrieve_meters(self):
+        """Retrieve the IDs of the meters used by Essent."""
+        return self._essent.get_EANs()
+
+
+class EssentMeter(Entity):
+    """Representation of Essent measurements."""
+
+    def __init__(self, username, password, meter, tariff):
+        """Initialize the sensor."""
+        self._state = None
+        self._username = username
+        self._password = password
+        self._meter = meter
+        self._tariff = tariff
+        self._meter_type = STATE_UNKNOWN
+        self._meter_unit = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return "Essent {} ({})".format(self._meter_type, self._tariff)
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        if self._meter_unit and self._meter_unit.lower() == 'kwh':
+            return ENERGY_KILO_WATT_HOUR
+
+        return self._meter_unit
+
+    def update(self):
+        """Fetch the energy usage."""
+        # Retrieve an authenticated session
+        essent = EssentBase(self._username, self._password).get_session()
```

Comment:
Must this object be created every update() ?

Reply:
I suppose not, but I don't know how long an authenticated session will last. I'd rather authenticate too often than have the component randomly fail because the authentication expired.

Should probably not update too often anyway as Essent will only update the statistics once a day. Is there a way to state that?

Reply:
You can use the `from homeassistant.util import Throttle` annotation. You can lookup other components that use this for an example.